server/app.es6

```diff
@@ -1,26 +1,391 @@
+      applicantInfoPrimaryMailingCity: req.body.applicantInfo.primaryAddress ? req.body.applicantInfo.primaryAddress.mailingCity : null,
+      applicantInfoPrimaryMailingState: req.body.applicantInfo.primaryAddress ? req.body.applicantInfo.primaryAddress.mailingState : null,
+      applicantInfoPrimaryMailingZIP: req.body.applicantInfo.primaryAddress ? req.body.applicantInfo.primaryAddress.mailingZIP : null,
+      applicantInfoSecondaryMailingAddress: req.body.applicantInfo.secondaryAddress ? req.body.applicantInfo.secondaryAddress.mailingAddress : null,
+      applicantInfoSecondaryMailingAddress2: req.body.applicantInfo.secondaryAddress ? req.body.applicantInfo.secondaryAddress.mailingAddress2 : null,
+      applicantInfoSecondaryMailingCity: req.body.applicantInfo.secondaryAddress ? req.body.applicantInfo.secondaryAddress.mailingCity : null,
+      applicantInfoSecondaryMailingState: req.body.applicantInfo.secondaryAddress ? req.body.applicantInfo.secondaryAddress.mailingState : null,
+      applicantInfoSecondaryMailingZIP: req.body.applicantInfo.secondaryAddress ? req.body.applicantInfo.secondaryAddress.mailingZIP : null,
+      applicantInfoOrganizationName: req.body.applicantInfo.organizationName,
+      applicantInfoWebsite: req.body.applicantInfo.website,
+      applicantInfoOrgType: req.body.applicantInfo.orgType,
+      applicantInfoSecondaryFirstName: req.body.applicantInfo.secondaryFirstName,
+      applicantInfoSecondaryLastName: req.body.applicantInfo.secondaryLastName,
+      type: req.body.type,
+      tempOutfitterFieldsIndividualCitizen: req.body.tempOutfitterFields.individualCitizen,
+      tempOutfitterFieldsSmallBusiness: req.body.tempOutfitterFields.smallBusiness,
+      tempOutfitterFieldsActivityDescription: req.body.tempOutfitterFields.activityDescription,
+      tempOutfitterFieldsAdvertisingUrl: req.body.tempOutfitterFields.advertisingURL,
+      tempOutfitterFieldsAdvertisingDescription: req.body.tempOutfitterFields.advertisingDescription,
+      tempOutfitterFieldsClientCharges: req.body.tempOutfitterFields.clientCharges,
+      tempOutfitterFieldsExperienceList: req.body.tempOutfitterFields.experienceList,
+      signature: req.body.signature,
+      reasonForReturn: req.body.reasonForReturn
+    }).then((tempOutfitterApp) => {
+      req.body['applicationId'] = tempOutfitterApp.applicationId;
+      req.body['appControlNumber'] = tempOutfitterApp.appControlNumber;
+      res.status(201).json(req.body);
+    }).error((err) => {
+      res.status(500).json(err);
+    });
+  }
+};
+
+// server creation ----------------------------------------------------------
 
 let app = express();
 app.use(bodyParser.json());
 
 // middleware that will add the Access-Control-Allow-Origin header to everything
 app.use(function(req, res, next) {
-  res.set('Access-Control-Allow-Origin', '*');
+  var origin = req.headers.origin;
+  if(origin === 'http://localhost:4200'){
```

Comment:
seems like you could use the env vars that are set for the frontend for this to make sure its consistent, because I could see a world that somehow that this origin could be faked.